chart/templates/secrets/result-backend-connection-secret.yaml

```diff
@@ -18,8 +18,10 @@
 ################################
 ## Airflow Result Backend Secret
 #################################
-{{- if (and .Values.data.resultBackendConnection (not .Values.data.resultBackendSecretName)) }}
+{{- if not .Values.data.resultBackendSecretName }}
+{{- if or (eq .Values.executor "CeleryExecutor") (eq .Values.executor "CeleryKubernetesExecutor") }}
```

Comment:
We were deploying this secret more often than we needed. This limits to only celery based executors.